core/jvm/src/JvmCore.kt

```diff
@@ -107,56 +103,62 @@ private open class InputStreamSource(
 
   override fun close() = input.close()
 
-  override fun cancel() {
-    // Not cancelable.
-  }
-
   override fun toString() = "source($input)"
 }
 
 /**
- * Returns a sink that writes to `socket`. Prefer this over [sink]
+ * Returns [RawSink] that writes to a socket. Prefer this over [sink]
  * because this method honors timeouts. When the socket
  * write times out, the socket is asynchronously closed by a watchdog thread.
+ *
+ * Use [RawSink.buffer] to create a buffered sink from it.
  */
-@Throws(IOException::class)
-fun Socket.sink(): RawSink {
-  return object : OutputStreamSink(getOutputStream()) {
-    override fun cancel() {
-      this@sink.close()
-    }
-  }
-}
+public fun Socket.sink(): RawSink = OutputStreamSink(getOutputStream())
```

Comment:
Not sure if I get the idea. Could you please elaborate?

Reply:
why we need `Socket.sink` as we can easily do `socket.outputStream.sink()`. Previously in Okio there was different implementation, which overrides `cancel`, now there is no such functionality.
Of course if there will be later introduced Async IO it can be revisited, though, I believe there are other java NIO primitives suited for Async IO (debatable of course).
This is just thoughts on reducing public API regarding what is **really** needed regarding interop, so this can be left as is for now

Reply:
Got it, thanks. Extensions for `j.i.File` and `j.n.Path` could be also removed. I'll consider that, thanks for the suggestion!

Reply:
https://github.com/Kotlin/kotlinx-io/issues/143